backend/src/Edna/Upload/Web/Types.hs

```diff
@@ -51,16 +51,29 @@ instance Buildable FileBS where
 instance ToSchema FileBS where
   declareNamedSchema _ = pure $ NamedSchema (Just "ByteString") binarySchema
 
-instance HasSwagger api => HasSwagger (MultipartForm Mem FileBS :> api) where
-  toSwagger _ = toSwagger (Proxy :: Proxy api) & addParam param
+instance HasOpenApi api => HasOpenApi (MultipartForm Mem FileBS :> api) where
+  toOpenApi _ = toOpenApi (Proxy :: Proxy api) & addRequestBody body
     where
-      param = mempty
-        & name .~ "file"
-        & required ?~ True
-        & description ?~ "Experiment to parse"
-        & schema .~ ParamOther (mempty
-            & in_ .~ ParamFormData
-            & paramSchema .~ (mempty & type_ ?~ S.SwaggerFile))
+      body :: O.RequestBody
+      body = mempty
+        & O.content .~ content
+        & O.required ?~ True
+
+      content :: Map.InsOrdHashMap MediaType O.MediaTypeObject
+      content = Map.singleton "multipart/form-data" $ mempty
```

Comment:
Instead of `Map.singleton` you can do

```haskell
content = mempty
  & at "multipart/form-data" ?~ mempty & schema ?~ paramSchema
```

Reply:
hm...
@maksbotan now I have a lot of errors:

```log
src/Edna/Upload/Web/Types.hs:63:26: error:
    • Could not deduce (Lens.Micro.Internal.At
                          (Map.InsOrdHashMap MediaType O.MediaTypeObject))
        arising from a use of ‘at’
      from the context: HasOpenApi api
        bound by the instance declaration
        at src/Edna/Upload/Web/Types.hs:54:10-71
    • In the first argument of ‘(?~)’, namely
        ‘at "multipart/form-data"’
      In the second argument of ‘(&)’, namely
        ‘at "multipart/form-data" ?~ mempty’
      In the first argument of ‘(&)’, namely
        ‘mempty & at "multipart/form-data" ?~ mempty’
   |
63 |       content = mempty & at "multipart/form-data" ?~ mempty
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^

src/Edna/Upload/Web/Types.hs:63:29: error:
    • Could not deduce (IsString
                          (Lens.Micro.Internal.Index
                             (Map.InsOrdHashMap MediaType O.MediaTypeObject)))
        arising from the literal ‘"multipart/form-data"’
      from the context: HasOpenApi api
        bound by the instance declaration
        at src/Edna/Upload/Web/Types.hs:54:10-71
    • In the first argument of ‘at’, namely ‘"multipart/form-data"’
      In the first argument of ‘(?~)’, namely ‘at "multipart/form-data"’
      In the second argument of ‘(&)’, namely
        ‘at "multipart/form-data" ?~ mempty’
   |
63 |       content = mempty & at "multipart/form-data" ?~ mempty
   |                             ^^^^^^^^^^^^^^^^^^^^^

src/Edna/Upload/Web/Types.hs:63:54: error:
    • Could not deduce (Monoid
                          (Lens.Micro.Internal.IxValue
                             (Map.InsOrdHashMap MediaType O.MediaTypeObject)))
        arising from a use of ‘mempty’
      from the context: HasOpenApi api
        bound by the instance declaration
        at src/Edna/Upload/Web/Types.hs:54:10-71
    • In the second argument of ‘(?~)’, namely ‘mempty’
      In the second argument of ‘(&)’, namely
        ‘at "multipart/form-data" ?~ mempty’
      In the first argument of ‘(&)’, namely
        ‘mempty & at "multipart/form-data" ?~ mempty’
   |
63 |       content = mempty & at "multipart/form-data" ?~ mempty
   |                                                      ^^^^^^

src/Edna/Upload/Web/Types.hs:65:11: error:
    • Could not deduce (O.HasSchema
                          (Map.InsOrdHashMap MediaType O.MediaTypeObject)
                          (Maybe (O.Referenced O.Schema)))
        arising from a use of ‘O.schema’
      from the context: HasOpenApi api
        bound by the instance declaration
        at src/Edna/Upload/Web/Types.hs:54:10-71
    • In the first argument of ‘(?~)’, namely ‘O.schema’
      In the second argument of ‘(&)’, namely ‘O.schema ?~ paramSchema’
      In the expression:
        mempty & at "multipart/form-data" ?~ mempty
          & O.schema ?~ paramSchema
   |
65 |         & O.schema ?~ paramSchema
   |           ^^^^^^^^
```

Reply:
That's what you get for using obscure packages like `microlens` :)
The required instances are present in `lens`. Can you clarify with @gromakovsky whether it's ok to use `Control.Lens.at` instead of microlens variant?

Reply:
> The required instances are present in `lens`. 

I think they are present somewhere else because `lens` doesn't seem to depend on `insert-ordered-containers`. But I don't want to search for these instances.

> whether it's ok to use `Control.Lens.at` instead of `microlens` variant?

Yes, I think it's ok to use anything from `lens` because we transitively depend on `lens` anyway.

Reply:
I meant they are defined for `Control.Lens.At`, not present in `lens`, sorry. Instances come from `insert-ordered-containers` and are transitively visible here from `openapi3` itself.